internal/services/monitor/monitor_diagnostic_setting_resource.go

```diff
@@ -318,7 +318,15 @@ func resourceMonitorDiagnosticSettingRead(d *pluginsdk.ResourceData, meta interf
 	}
 
 	d.Set("name", id.Name)
-	d.Set("target_resource_id", id.ResourceUri)
+
+	resourceId, err := azure.ParseAzureResourceID(actualResourceId)
+	if err != nil {
+		return err
+	}
+	if resourceId.ResourceGroup != "" {
+		actualResourceId = strings.Replace(actualResourceId, "/resourcegroups/", "/resourceGroups/", 1)
+	}
+	d.Set("target_resource_id", actualResourceId)
```

Comment:
rather than parsing the Resource ID we should:

1. Add a state migration to conditionally string replace `resourcegroups` with `resourceGroups` as appropriate
2. Not parse the Resource ID here but instead string replace `resourcegroups` with `resourceGroups` as appropriate

Reply:
I think we should not use Strings.Replace at all, a corner case is if **the Name of the resource group (or other segment) is `resourcegroups`** , the resourceId is like`/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resourcegroups/providers/Microsoft.KeyVault/vaults/vault1|logMonitoring1"` ,  then if we should not replace it with `resourceGroups`.